app/src/trackPlayer.ts

```diff
@@ -0,0 +1,78 @@
+import RNTrackPlayer, { Track, Capability, Event, State } from 'react-native-track-player'
+
+export const initialPlayerState = {
+    currentTrack: null,
+    currentIndex: null,
+    recordsCount: 0,
+    isPlaying: false
+}
+
+export async function createPlayerState(event: {
+    type: Event;
+    [key: string]: any;
+}) {
+    const queue = await RNTrackPlayer.getQueue()
+    const currentIndex = await RNTrackPlayer.getCurrentTrack()
+    const state = await RNTrackPlayer.getState();
+    const track = await RNTrackPlayer.getTrack(currentIndex)
+
+    return {
+        currentTrack: track,
+        currentIndex: currentIndex,
+        recordsCount: queue.length,
+        isPlaying: state == State.Playing
```

Comment:
mozna bych ten `state` ukladal primo, a odvodil si `isPlaying` az kdyz to budu renderovat... neco mi rika, ze se nam ty dalsi stavy mohou v budoucnu hodit

Reply:
coz ted kdyz po sobe ctu asi zni jako premature optimization, ale asi je to i lepsi mit jako `state`

Reply:
je to prototyp, stale neznam mnoho odpovedi o tom, jak to bude fungovat